test_hook_writes_to_stderr failure "/bin/sh command not found"

### Subject of the issue
`FAILED tests/test_hooks/test_cmd.py::test_hook_writes_to_stderr - AssertionError: assert '/bin/sh: mis...and not found' == '/bin/sh: 1: ...nd: not found'` when I run Sceptre unit test.


### Your environment
* Sceptre v 4.3.0
* Python v 3.9
* MAC OSX Sonoma 14.0

### Steps to reproduce

1. poetry install
2. pyenv shell 3.9
3. ➜  python -V
Python 3.9.7
4. ➜  /bin/sh --version
GNU bash, version 3.2.57(1)-release (x86_64-apple-darwin23)
Copyright (C) 2007 Free Software Foundation, Inc.
5. poetry run pytest


### Expected behaviour
All tests pass


### Actual behaviour
1 test failure..

```
stack = sceptre.stack.Stack(name='stack1', project_code=project1, template_handler_config={'template': 'path.yaml'}, region=re...ions=[], on_failure=None, disable_rollback=False, stack_timeout=0, stack_group_config={}, ignore=False, obsolete=False)
capfd = <_pytest.capture.CaptureFixture object at 0x1102a76d0>

    def test_hook_writes_to_stderr(stack, capfd):
        with pytest.raises(Exception):
            Cmd("missing_command", stack).run()
        cap = capfd.readouterr()
        assert cap.out.strip() == ""
>       assert cap.err.strip() == "/bin/sh: 1: missing_command: not found"
E       AssertionError: assert '/bin/sh: mis...and not found' == '/bin/sh: 1: ...nd: not found'
E         - /bin/sh: 1: missing_command: not found
E         ?         ---
E         + /bin/sh: missing_command: command not found
E         ?                           ++++++++

tests/test_hooks/test_cmd.py:150: AssertionError
```

--- a/tests/test_hooks/test_cmd.py
+++ b/tests/test_hooks/test_cmd.py
@@ -138,11 +138,10 @@ def test_hook_writes_to_stdout(stack, capfd):
 
 
 def test_hook_writes_to_stderr(stack, capfd):
-    with pytest.raises(Exception):
-        Cmd("missing_command", stack).run()
+    Cmd("echo hello >&2", stack).run()
     cap = capfd.readouterr()
     assert cap.out.strip() == ""
-    assert cap.err.strip() == "/bin/sh: 1: missing_command: not found"
+    assert cap.err.strip() == "hello"
 
 
 def test_default_shell_is_sh(stack, capfd):

